foreshadow/cleaners/data_cleaner.py

```diff
@@ -79,22 +101,44 @@ def pick_transformer(self, X, y=None, **fit_params):
         best_score = 0
         best_cleaner = None
         for cleaner, name in cleaners:
-            cleaner = cleaner(column_sharer=self.column_sharer, name=name)
+            cleaner = cleaner(column_sharer=self.column_sharer)
             score = cleaner.metric_score(X)
             if score > best_score:
                 best_score = score
                 best_cleaner = cleaner
-
         if best_cleaner is None:
             return NoTransform(column_sharer=self.column_sharer)
         return best_cleaner
 
+    def resolve(self, X, *args, **kwargs):
+        """Resolve the underlying concrete transformer.
+
+        Sets self.column_sharer with the domain tag.
+
+        Args:
+            X: input DataFrame
+            *args: args to super
+            **kwargs: kwargs to super
+
+        Returns:
+            Return from super.
+
+        """
+        ret = super().resolve(X, *args, **kwargs)
+        if self.column_sharer is not None:
+            self.column_sharer[
+                "domain", X.columns[0]
+            ] = self.transformer.__class__.__name__
+        else:
+            logging.debug("column_sharer was None")
```

Comment:
Is ColumnSharer ever going to be None in DataCleaner? If not, should we through exceptions instead?

Reply:
It could be -> you may want to clean data without the rest of the pipeline. In this case you would not use the Foreshadow object and just use the DataCleaner object (this paradigm is similar across all of the steps in PreparerSteps). If we don't want this use case we should briefly discuss its importance